cloud/deploy/deploy.go

```diff
@@ -190,15 +193,24 @@ func Deploy(deployInput InputDeploy, client astro.Client) error { //nolint
 	deploymentURL := "cloud." + domain + "/" + deployInfo.workspaceID + "/deployments/" + deployInfo.deploymentID + "/analytics"
 
 	if deployInput.Dags {
-		if deployInput.Pytest == allTests {
+		if (deployInput.Pytest == allTests || deployInput.Parse) && !deployInput.ForceDeploy {
 			version, err := buildImage(&c, deployInput.Path, deployInfo.currentVersion, deployInfo.deployImage, deployInput.ImageName, deployInfo.dagDeployEnabled, client)
 			if err != nil {
 				return err
 			}
 
-			err = parseDAG(deployInput.Pytest, version, deployInput.EnvFile, deployInfo.deployImage, deployInfo.namespace)
-			if err != nil {
-				return err
+			if deployInput.Pytest == allTests {
+				err = parseOrPytestDAG(deployInput.Pytest, version, deployInput.EnvFile, deployInfo.deployImage, deployInfo.namespace)
+				if err != nil {
+					return err
+				}
+			}
+
+			if deployInput.Parse {
+				err = parseOrPytestDAG("parse", version, deployInput.EnvFile, deployInfo.deployImage, deployInfo.namespace)
```

Comment:
```suggestion
				err = parseOrPytestDAG(parse, version, deployInput.EnvFile, deployInfo.deployImage, deployInfo.namespace)
```
nitpick: Reusing already defined constant in the package